eclipse-collections-api/src/main/java/org/eclipse/collections/api/map/ImmutableOrderedMap.java

```diff
@@ -159,4 +158,5 @@
 
     @Override
     <KK, VV> ImmutableOrderedMap<KK, VV> aggregateBy(Function<? super V, ? extends KK> groupBy, Function0<? extends VV> zeroValueFactory, Function2<? super VV, ? super V, ? extends VV> nonMutatingAggregator);
+    */
```

Comment:
Thanks for pointing this out. The answer is a little different here. ImmutableCollection defined these methods as returning ImmutableMap. The backwards compatible thing to do is say ImmutableIterable's aggregation methods return ImmutableMapIterable and ImmutableCollection overrides that to ImmutableMap. But even that isn't right, because ImmutableSortedSet should return an Ordered map. But Ordered maps don't exist yet, so there's probably no point in breaking backwards compatibility yet.

So yes, this is a TODO for when concrete OrderedMaps exist. However, I should also probably put the overrides back into MutableCollection and ImmutableCollection in this pull request. I'll add a second commit and if it looks good to you, squash it.